ios/brave-ios/Sources/AIChat/ModelView/AIChatViewModel.swift

```diff
@@ -0,0 +1,305 @@
+// Copyright 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import Foundation
+import SwiftUI
+import WebKit
+import BraveCore
+import Shared
+
+public class AIChatViewModel: NSObject, ObservableObject {
+  private var api: AIChat!
+  private let webView: WKWebView?
```

Comment:
Does this need to be weak?

Reply:
It's not really possible for the WebView to be closed while the AIChat window is open, but I made it weak anyway even though it deinit just fine :)